test/unit/meta/concepts/scalar.cpp

```diff
@@ -82,8 +79,6 @@ TTS_CASE("Check for arithmetic_scalar_value on plain_scalar_value" )
   TTS_EXPECT( eve::arithmetic_scalar_value<std::uint16_t>  );
   TTS_EXPECT( eve::arithmetic_scalar_value<std::uint32_t>  );
   TTS_EXPECT( eve::arithmetic_scalar_value<std::uint64_t>  );
-  TTS_EXPECT( eve::arithmetic_scalar_value<std::size_t>    );
```

Comment:
I need to get more info on what Apple Clang doesn't liek about those. 
size_t and ptrdiff_t are common type people will want to use.

Reply:
So is char. But it doesn't work.

If we want to support all of the weird types that are not actually the same, we have to solve it as a separate problem.